### ISSUE TITLE
the accuracy was very low

### ISSUE THREAD

**Issue by wenxian-ok** [nishadsinghi/CleanCLIP#12]
I used the following command to pre-train the CLIP model:  
`python -m src.main --name Pre_Training_badnet_nopre --train_data /data/New_User/Wmw/Backdoor_Attack/Data/CC3M_big/backdoor_banana_random_random_16_500000_1500.csv --validation_data /data/New_User/Wmw/Backdoor_Attack/Data/CC3M_big/val_dataset.csv --image_key image --caption_key caption --device_id 0 --pretrained`  

This loaded the CLIP pre-trained model. After obtaining the poisoned `.pt` file, I fine-tuned it using 100,000 pairs of clean image-text data with the command:  
`python -m src.main --name cleanCLIP_finetuning_badnet --checkpoint /data/New_User/Wmw/Backdoor_Attack/Defense/CleanCLIP-main/logs/Pre_Training_badnet/checkpoints/epoch.best.pt --device_id 0 --batch_size 64 --train_data /data/New_User/Wmw/Backdoor_Attack/Data/CC3M_big/finetune.csv --epochs 10 --num_warmup_steps 50 --lr 1e-5 --inmodal --complete_finetune`  

However, when validating on the ImageNet1k dataset, the accuracy was very low：

![Image](https://github.com/user-attachments/assets/21a9566b-4fb2-486e-8560-f501e7f518df)

May I ask what the reason for this is?

**Comment by nishadsinghi** [comment]
In the paper, we pre-trained the models on CC3M, i.e., 3M samples, which led to 19% accuracy. Looks like you're pre-training on 500K samples. Since this is significantly lower than 3M, your accuracy of 9.9% is not that surprising. I would suggest using 3M samples to get similar results to the paper.

**Comment by wenxian-ok** [comment]
> In the paper, we pre-trained the models on CC3M, i.e., 3M samples, which led to 19% accuracy. Looks like you're pre-training on 500K samples. Since this is significantly lower than 3M, your accuracy of 9.9% is not that surprising. I would suggest using 3M samples to get similar results to the paper.

Thank you very much for your reply. However, I have a question. When I pre-trained with 500K samples, I used the `--pretrained` parameter, which means the original CLIP pre-trained weights were loaded. I would like to know whether you enabled this `--pretrained` parameter when pre-training with 3M samples.

**Comment by nishadsinghi** [comment]
In the paper, we used these hyperparameters to poison a pre-trained model:

> We finetune the pretrained model for 5 epochs with an initial learning rate of 1e-6 with cosine scheduling
> and 50 warmup steps, and use AdamW as the optimizer.

Can you try this and then evaluate its clean accuracy and ASR? (without removing the backdoor)

**Comment by wenxian-ok** [comment]
> In the paper, we used these hyperparameters to poison a pre-trained model:
> 
> > We finetune the pretrained model for 5 epochs with an initial learning rate of 1e-6 with cosine scheduling
> > and 50 warmup steps, and use AdamW as the optimizer.
> 
> Can you try this and then evaluate its clean accuracy and ASR? (without removing the backdoor)

Could you provide me with a command or instructions for fine-tuning a pre-trained model?

**Comment by nishadsinghi** [comment]
It's been a while since we worked on this, so my memory might not be perfect, but try appending this to your command:
`--epochs 5 --num_warmup_steps 50 --lr 1e-6`
Also, try evaluating the clean accuracy and ASR of the pretrained model without any fine-tuning. This way you'll know if the evaluation command is working properly.
